jerry-core/api/jerry.c

```diff
@@ -3311,24 +3289,10 @@ jerry_property_descriptor_to_ecma (const jerry_property_descriptor_t *prop_desc_
 {
   ecma_property_descriptor_t prop_desc = ecma_make_empty_property_descriptor ();
 
-  uint32_t flags = ECMA_PROP_NO_OPTS;
-
-  if (prop_desc_p->is_enumerable_defined)
-  {
-    flags |= (uint32_t) (ECMA_PROP_IS_ENUMERABLE_DEFINED | (prop_desc_p->is_enumerable ? ECMA_PROP_IS_ENUMERABLE
-                                                                                       : ECMA_PROP_NO_OPTS));
-  }
-
-  if (prop_desc_p->is_configurable_defined)
-  {
-    flags |= (uint32_t) (ECMA_PROP_IS_CONFIGURABLE_DEFINED | (prop_desc_p->is_configurable ? ECMA_PROP_IS_CONFIGURABLE
-                                                                                           : ECMA_PROP_NO_OPTS));
-  }
+  prop_desc.flags = prop_desc_p->flags;
```

Comment:
If we are expecting that the internal prop flags are the same as the public api ones, we need to have a static assert to validated they have the same value.